### PR TITLE
add: `os-installer-git`

### PR DESCRIPTION
**Pull request by cat-master21**

### CODE REVIEW

**Comment by cat-master21** [comment]
works on ubuntu Kinetic and Debian Sid only https://github.com/cat-master21/pacstall-programs/actions/runs/3401189735/jobs/5656130607 due to libgnome-desktop having different library names.

**Comment by Henryws** [comment]
Can you make a case statement for each distros depends?

**Comment by cat-master21** [comment]
I'd rather not as the amount of distros that it would support keeps changing (like adding lunar and bookworm to stable soon). If the package isn't avaible then it would just fail anyway @Henryws.

**Comment by Henryws** [comment]
Ok then at least add a warning that tells the user that this package is intended for a very specific use-case, and should not be installed unless they know why they need to.

File: packages/os-installer-git/os-installer-git.pacscript [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+version="$(pkgver)"
+
+prepare() {
+  git submodule update --init --recursive
+  fancy_message warn "This can only be built using the latest version of most OSes."
+}
+
+build() {
+  # Build nessesary files
+  meson --prefix="${STOWDIR}/${name}/usr" --buildtype=release . build
```
Comment (Henryws) [inline packages/os-installer-git/os-installer-git.pacscript:25]:
```suggestion
  meson --prefix="${pkgdir}/usr" --buildtype=release . build
```

File: packages/os-installer-git/os-installer-git.pacscript [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+install() {
+  # Install license
+  sudo install -Dm644 "LICENSE" doc/* -t "${pkgdir}/usr/share/licenses/${pkgname}"
+
+  # Install doc
+  sudo install -Dm644 "README.md" -t "${pkgdir}/usr/share/doc/${pkgname}"
+
+  # Install
+  sudo meson install -C build --destdir "${STOWDIR}/${name}/usr"
```
Comment (Henryws) [inline packages/os-installer-git/os-installer-git.pacscript:36]:
```suggestion
  sudo meson install -C build --destdir "${pkgdir}/usr"
```

**Review by Henryws — CHANGES_REQUESTED** [review]

**Comment by Henryws** [comment]
Hmm, could this actually be put into a separate repository, or could it stay here..

**Comment by cat-master21** [comment]
Well I'd rather it be here so that it can be accessible to other users more easily and if your refering to Rhino Linux's repo I heard that we should put the pacscripts here as you are still maintianing both and it's just less work generally.

**Review by Henryws — APPROVED** [review]

**Pull request merged**